crates/bevy_animation/src/lib.rs

```diff
@@ -138,68 +149,109 @@ pub enum Interpolation {
+/// [`AnimationTargetId`]. Thus a bone named `Chest` directly connected to a
+/// bone named `Hips` will have a different ID from a bone named `Chest` that's
+/// connected to a bone named `Stomach`.
+///
+/// [UUID]: https://en.wikipedia.org/wiki/Universally_unique_identifier
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Reflect, Debug)]
+pub struct AnimationTargetId(pub Uuid);
+
+/// An entity that can be animated by an [`AnimationPlayer`].
+///
+/// These are frequently referred to as *bones* or *joints*, because they often
+/// refer to individually-animatable parts of an armature.
+///
+/// Asset loaders for armatures are responsible for adding these as necessary.
+/// Typically, they're generated from hashed versions of the entire name path
+/// from the root of the armature to the bone. See the [`AnimationTargetId`]
+/// documentation for more details.
+///
+/// By convention, asset loaders add [`AnimationTarget`] components to the
+/// descendants of an [`AnimationPlayer`], but Bevy doesn't require this in any
+/// way. So, for example, it's entirely possible for an [`AnimationPlayer`] to
+/// animate a target that it isn't an ancestor of. If you add a new bone to or
+/// delete a bone from an armature at runtime, you may want to update the
+/// [`AnimationTarget`] component as appropriate, as Bevy won't do this
+/// automatically.
+///
+/// Note that each entity can only be animated by one animation player at a
+/// time. However, you can change [`AnimationTarget`]'s `player` property at
+/// runtime to change which player is responsible for animating the entity.
+#[derive(Clone, Component, Reflect)]
+#[reflect(Component, MapEntities)]
+pub struct AnimationTarget {
+    /// The ID of this animation target.
+    ///
+    /// Typically, this is derived from the path.
+    pub id: AnimationTargetId,
+
+    /// The entity containing the [`AnimationPlayer`].
+    pub player: Entity,
+}
+
 impl AnimationClip {
     #[inline]
-    /// [`VariableCurve`]s for each bone. Indexed by the bone ID.
-    pub fn curves(&self) -> &Vec<Vec<VariableCurve>> {
+    /// [`VariableCurve`]s for each animation target. Indexed by the [`AnimationTargetId`].
+    pub fn curves(&self) -> &AnimationCurves {
         &self.curves
     }
 
     /// Gets the curves for a bone.
     ///
-    /// Returns `None` if the bone is invalid.
-    #[inline]
-    pub fn get_curves(&self, bone_id: usize) -> Option<&'_ Vec<VariableCurve>> {
-        self.curves.get(bone_id)
-    }
-
-    /// Gets the curves by it's [`EntityPath`].
-    ///
-    /// Returns `None` if the bone is invalid.
+    /// Returns `None` if this clip doesn't animate this bone.
     #[inline]
-    pub fn get_curves_by_path(&self, path: &EntityPath) -> Option<&'_ Vec<VariableCurve>> {
-        self.paths.get(path).and_then(|id| self.curves.get(*id))
+    pub fn get_curves(&self, bone_id: AnimationTargetId) -> Option<&'_ Vec<VariableCurve>> {
```

Comment:
I don't have a good suggestion, but having both `curves()` and `get_curves()` (both plural) is quite confusing from an API perspective. Maybe rename to `get_target_curves()` this one? Or have two types `AnimationCurves` (all targets) vs. `AnimationTargetCurves` (one target)?

Reply:
I renamed it to `curves_for_target()`.